core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -143,9 +154,24 @@ static void prepareHeaders(Metadata headers, CallOptions callOptions, String use
   }
 
   @Override
-  public void start(Listener<RespT> observer, Metadata headers) {
+  public void start(final Listener<RespT> observer, Metadata headers) {
     Preconditions.checkState(stream == null, "Already started");
 
+    if (context.isCancelled()) {
+      // Context is already cancelled, cancel immediately and notify the observer of cancellation
+      // via callback on the executor
+      cancel();
```

Comment:
Why call `cancel()`? I don't see anything there that is necessary, since the call wasn't started. IIRC `cancelCalled` is just there as an optimization to prevent trying to RST_STREAM a ton of times.